python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -127,18 +132,17 @@ def read_partition(
         if partition_keys:
             if partitions is None:
                 raise ValueError("Must pass partition sets")
+
             for i, (name, index2) in enumerate(partition_keys):
-                categories = [
-                    val.as_py() for val in partitions.levels[i].dictionary
-                ]
+
+                categories = partitions[i].keys
 
                 col = as_column(index2).as_frame().repeat(len(df))._data[None]
-                df[name] = build_categorical_column(
-                    categories=categories,
-                    codes=as_column(col.base_data, dtype=col.dtype),
-                    size=col.size,
-                    offset=col.offset,
-                    ordered=False,
+
```

Comment:
A lot nicer with supported categorical column creation!

Reply:
I just realized that the new change here was a bit inefficent.  We were creating a column with the partition-based category repeated in every element, and then converting it to a categorical column.  It makes a bit more sense to repeat the index of the partition-based category in every element, and build the categorical column directly.